packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -52,6 +52,105 @@ async function offerToCreateTemplate(): Promise<void> {
   }
 }
 
+function findFoamVariables(templateText: string): string[] {
+  const regex = /\$(FOAM_[_a-zA-Z0-9]*)|\${(FOAM_[[_a-zA-Z0-9]*)}/g;
+  var matches = [];
+  const output: string[] = [];
+  while ((matches = regex.exec(templateText))) {
+    output.push(matches[1] || matches[2]);
+  }
+  const uniqVariables = [...new Set(output)];
+  return uniqVariables;
+}
+
+function getFoamTitle(givenValues: Map<string, string>) {
+  if (givenValues.has('FOAM_TITLE')) {
+    return Promise.resolve(givenValues.get('FOAM_TITLE'));
+  } else {
+    return window.showInputBox({
+      prompt: `Enter a title for the new note`,
+      value: 'Title of my New Note',
+      validateInput: value =>
+        value.trim().length === 0 ? 'Please enter a title' : undefined,
+    });
+  }
+}
+
+function slugify(input: string) {
+  // TODO: Replace this with the actual algorithm.
+  return input.toLowerCase().replace(/[^\p{L}\p{N}]/gu, '-');
+}
+
+async function getFoamTitleSlug(
+  resolver: Resolver,
+  givenValues: Map<string, string>
+) {
+  if (givenValues.has('FOAM_TITLE_SLUG')) {
```

Comment:
mmmm... why not always derive it from the `FOAM_TITLE`?

Reply:
For symmetry of implementation between all the variable resolver methods.

In fact, now that my code is cleaner than what I previously had, I had planned to pull this check up into the `resolve` method. Something like:

```javascript
if (givenValues.has(name)) {
  return Promise.resolve(givenValues.get(name))
} else {
  switch(name) {
     ...
  }
}
```

Reply:
Done.